File: solutions/c/grains/1/grains.c
#include "grains.h"
#include <math.h>

uint64_t square(uint8_t index) {
    if(index > 64) {
        return 0;
    }
    return pow(2, index - 1);
}

uint64_t total(void) {
    int total = 0;
    for(int x = 1; x <= 64; x++) {
        total += square(x);
    }
    return total;
}

File: solutions/java/grains/1/src/main/java/Grains.java
import java.math.BigInteger;

class Grains {

    BigInteger grainsOnSquare(final int square) {
        if (square < 1 || square > 64) {
            throw new IllegalArgumentException("square must be between 1 and 64");
        }
        return BigInteger.TWO.pow(square - 1);
    }

    BigInteger grainsOnBoard() {
        BigInteger sum = BigInteger.ZERO;
        for (int x = 1; x <= 64; x++) {
            sum = sum.add(grainsOnSquare(x));
        }
        return sum;
    }

}
